build/flaws.js

```diff
@@ -15,159 +15,188 @@ const {
-              a,
-              checked.get(href),
-              href,
-              found.url + absoluteURL.search + absoluteURL.hash
-            );
-          }
+          addBrokenLink(a, href);
+        }
+      } else {
+        // But does it have the correct case?!
+        if (found.url !== href.split("#")[0]) {
+          // Inconsistent case.
+          addBrokenLink(
+            a,
+            checked.get(href),
+            href,
+            found.url + absoluteURL.search + absoluteURL.hash
+          );
         }
       }
-    });
-    if (options.flawLevels.get("broken_links") === FLAW_LEVELS.ERROR) {
-      throw new Error(`broken_links flaws: ${doc.flaws.broken_links}`);
     }
+  });
+  if (
+    level === FLAW_LEVELS.ERROR &&
+    doc.flaws.broken_links &&
+    doc.flaws.broken_links.length
+  ) {
+    throw new Error(
+      `broken_links flaws: ${doc.flaws.broken_links.map(JSON.stringify)}`
+    );
   }
+}
 
-  if (options.flawLevels.get("bad_bcd_queries") !== FLAW_LEVELS.IGNORE) {
-    $("div.bc-data").each((i, element) => {
-      const dataQuery = $(element).attr("id");
-      if (!dataQuery) {
+// Bad BCD queries are when the `<div class="bc-data">` tags have an
+// ID (or even lack the `id` attribute) that don't match anything in the
+// @mdn/browser-compat-data package. E.g. Something like this:
+//
+//    <div class="bc-data" id="bcd:never.ever.heard.of">
+//
+function injectBadBCDQueriesFlaws(level, doc, $) {
+  if (level === FLAW_LEVELS.IGNORE) return;
+
+  $("div.bc-data").each((i, element) => {
+    const dataQuery = $(element).attr("id");
+    if (!dataQuery) {
+      if (!("bad_bcd_queries" in doc.flaws)) {
+        doc.flaws.bad_bcd_queries = [];
+      }
+      doc.flaws.bad_bcd_queries.push({
+        id: `bad_bcd_queries${doc.flaws.bad_bcd_queries.length}`,
+        explanation: "BCD table without an ID",
+        suggestion: null,
+      });
+    } else {
+      const query = dataQuery.replace(/^bcd:/, "");
+      const { data } = packageBCD(query);
+      if (!data) {
         if (!("bad_bcd_queries" in doc.flaws)) {
           doc.flaws.bad_bcd_queries = [];
         }
         doc.flaws.bad_bcd_queries.push({
           id: `bad_bcd_queries${doc.flaws.bad_bcd_queries.length}`,
-          explanation: "BCD table without an ID",
+          explanation: `No BCD data for query: ${query}`,
           suggestion: null,
         });
-      } else {
-        const query = dataQuery.replace(/^bcd:/, "");
-        const { data } = packageBCD(query);
-        if (!data) {
-          if (!("bad_bcd_queries" in doc.flaws)) {
-            doc.flaws.bad_bcd_queries = [];
-          }
-          doc.flaws.bad_bcd_queries.push({
-            id: `bad_bcd_queries${doc.flaws.bad_bcd_queries.length}`,
-            explanation: `No BCD data for query: ${query}`,
-            suggestion: null,
-          });
-        }
       }
-    });
-    if (options.flawLevels.get("broken_links") === FLAW_LEVELS.ERROR) {
```

Comment:
This was the old bug. It looked to see if the flaw level for `broken_links` was set to error in the context of checking `bad_bcd_queries`.